dts/bindings/input/zephyr,input-lvgl-pointer.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  LVGL pointer indev pseudo-device
+
+  Listens for touch input events and routes the
+  lv_indev_data_t to the underlying pointer lv_indev_t managed by LVGL.
+
+  Needs to be associated to a specific device to listen for events
+  from that device. Example configuration:
+
+  pointer {
+          compatible = "zephyr,input-lvgl-pointer";
+          input = <&input_sdl_touch>;
+  };
+
+compatible: "zephyr,input-lvgl-pointer"
```

Comment:
I'm a bit concerned about bindings specific to 3rd party modules. I think we need to avoid this.

Reply:
@mbolivar-ampere @galak

Reply:
Yeah, this seems potentially borderline. Does it comply with the rules in https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#the-zephyr-prefix or is it violating them somehow?

Reply:
How about using an `lvgl,` prefix for these? Would make them more compact as well, `lvgl,input-pointer`.

Anyway I think @gmarull was referring to the fact that the binding is for a module, not the name itself. I think it's fine in this case (or better, I don't see a better alternative) but yeah worth discussing.

Reply:
Well, actually I see other precedents (openamp):

> You may use the zephyr, prefix when naming a devicetree compatible that is specific to Zephyr. One example is [zephyr,ipc-openamp-static-vrings](https://docs.zephyrproject.org/latest/build/dts/api/bindings/ipc/zephyr%2Cipc-openamp-static-vrings.html#std-dtcompatible-zephyr-ipc-openamp-static-vrings). In this case, it’s permitted but not required to add the zephyr, prefix to properties defined in the binding.

So no objections. I wonder though if something like this is LVGL specific or could be made generic for other libraries.

Reply:
I think an `lvgl` prefix is probably worse than `zephyr` -- we have reasonable expectation that the project is big enough that the zephyr prefix is "ours" at this point, kind of like how linux has sometimes at least historically used `linux` as a prefix. So we can hide our mistakes in our namespace 😉 

If this can be made generic, all the better, but that sounds like a separate question. Overall my conclusion is that using the zephyr prefix here is OK.